src/ast.ml

```diff
@@ -639,15 +641,17 @@ and
       ((stmt list * loc (* Close brace *)) * int (*rank*)) option * 
       method_binding * 
       visibility *
-      bool (* is declared abstract? *)
+      bool * (* is declared abstract? *)
+      (string * ghostness) list (*type parameters*)

```

Comment:
I would suggest to only add support for real type parameters for now, not ghost ones. (Actually, in the interest of having minimal PRs, it would perhaps have been even better to not yet support type parameters on methods at all for now.)